pkg/rules/rules_test.go

```diff
@@ -0,0 +1,126 @@
+			{URI: "/path", ID: 9011, RuleTemplate: "# Some other template"},
+			{URI: "/path", ID: 9012},
+			{URI: "/path", ID: 942430, Description: `ModSecurity: Warning. Pattern match "((?:[~!@#\\$%\\^&\\*\\(\\)\\-\\+=\\{\\}\\[\\]\\|:;\"'\xc2\xb4\xe2\x80\x99\xe2\x80\x98<>][^~!@#\\$%\\^&\\*\\(\\)\\-\\+=\\{\\}\\[\\]\\|:;\"'\xc2\xb4\xe2\x80\x99\xe2\x80\x98<>]*?){6})" at ARGS:variables.`},
+		},
+		`
+# Path /path
+# This is the default template
+# Some other template
+SecRule REQUEST_URI "@strmatch /path" \
+  "phase:2,nolog,id:10101010,\
+  ctl:ruleRemoveById=9010,\
+  ctl:ruleRemoveById=9011,\
+  ctl:ruleRemoveById=9012,\
+  ctl:ruleRemoveTargetById=942430;ARGS:variables"
+`,
+	},
+	{
+		"Multiple paths",
+		[]model.ModsecAlert{
+			{URI: "/path/one", ID: 9010, RuleTemplate: "# This is the default template"},
+			{URI: "/path/two", ID: 9011, RuleTemplate: "# Some other template"},
+			{URI: "/path/three", ID: 9012},
+			{URI: "/some/path", ID: 942430, Description: "ModSecurity: Warning. Pattern match W{4} at ARGS:query.", RuleTemplate: "# ModSec Rule Exclusion: 942430 : Restricted SQL Character Anomaly Detection (args): # of special characters exceeded (12) (severity:  WARNING) PL2"},
+		},
+		`
+# This is the default template
+SecRule REQUEST_URI "@strmatch /path/one" \
+  "phase:2,nolog,id:10101010,\
+  ctl:ruleRemoveById=9010"
+
+SecRule REQUEST_URI "@strmatch /path/three" \
+  "phase:2,nolog,id:10101011,\
+  ctl:ruleRemoveById=9012"
+
+# Some other template
+SecRule REQUEST_URI "@strmatch /path/two" \
+  "phase:2,nolog,id:10101012,\
+  ctl:ruleRemoveById=9011"
+
+# ModSec Rule Exclusion: 942430 : Restricted SQL Character Anomaly Detection (args): # of special characters exceeded (12) (severity:  WARNING) PL2
+SecRule REQUEST_URI "@strmatch /some/path" \
+  "phase:2,nolog,id:10101013,\
+  ctl:ruleRemoveTargetById=942430;ARGS:query"
+`,
+	},
+	{
+		"With parameter",
+		[]model.ModsecAlert{
+			{URI: "/some/path", ID: 942430, Description: "ModSecurity: Warning. Pattern match W{4} at ARGS:query."},
+			{URI: "/some/path", ID: 942431, Description: `ModSecurity: Warning. Pattern match "((?:[~!@#\\$%\\^&\\*\\(\\)\\-\\+=\\{\\}\\[\\]\\|:;\"'\xc2\xb4\xe2\x80\x99\xe2\x80\x98<>][^~!@#\\$%\\^&\\*\\(\\)\\-\\+=\\{\\}\\[\\]\\|:;\"'\xc2\xb4\xe2\x80\x99\xe2\x80\x98<>]*?){6})" at ARGS:identificationRedirectURL.`},
+		},
+		`
+# Path /some/path
+SecRule REQUEST_URI "@strmatch /some/path" \
+  "phase:2,nolog,id:10101010,\
+  ctl:ruleRemoveTargetById=942430;ARGS:query,\
+  ctl:ruleRemoveTargetById=942431;ARGS:identificationRedirectURL"
+`,
+	},
+	{
+		"Non matched parameter",
+		[]model.ModsecAlert{
+			{URI: "/", ID: 921180, Description: `ModSecurity: Warning. Pattern match "TX:paramcounter_(.*)" at TX:paramcounter_ARGS_NAMES:prospectSingle.contactMethods.contactMethods.value.`},
+		},
+		`
+SecRule REQUEST_URI "@strmatch /" \
+  "phase:2,nolog,id:10101010,\
+  ctl:ruleRemoveById=921180"
+`,
+	},
+}
```

Comment:
This is a bit cumbersome to read... would it make sense to put the rules in separate files and read them from the `testdata` dir?

Reply:
Good idea!